### PR TITLE
chore(core): include @aws-solutions-konstruk in version reporting

### LINKED ISSUE #7365: Include AWS Solutions Konstruk constructs usage in AWS::CDK::Metadata version reporting
Augment AWS CDK built-in version reporting mechanism to include the usage of AWS Solutions Konstruk constructs (@aws-solutions-konstruk/*)

### Use Case
[AWS Solutions Team](https://aws.amazon.com/solutions/) has published suite of CDK L3 Constructs on NPM [here](https://www.npmjs.com/org/aws-solutions-konstruk) and would like to leverage the "AWS::CDK::Metadata" version reporting mechanism to get the usage metrics of AWS Solutions Konstruk constructs.

### Proposed Solution

Submit the PR request against the following code & documentation

1 – Code: https://github.com/aws/aws-cdk/blob/8c6fc43fab73d72e2ad03013635ed21e830686fb/packages/%40aws-cdk/core/lib/private/runtime-info.ts#L18-L22
2 – Documentation: https://github.com/awsdocs/aws-cdk-guide/blob/master/doc_source/tools.md#version-reporting

* [ X] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change
---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by hnishar**
### Commit Message
chore(core): include @aws-solutions-konstruk in version reporting (#7394)

Since this is an AWS-vended module, we will include it in our module version analytics.

Closes #7365

### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/core/lib/private/runtime-info.ts [hunk 1]
```diff
@@ -14,9 +14,9 @@ export function collectRuntimeInformation(): cxschema.RuntimeInfo {
     }
   }
 
-  // include only libraries that are in the @aws-cdk npm scope
+  // include only libraries that are in the @aws-cdk and @aws-solutions-konstruk npm scopes
   for (const name of Object.keys(libraries)) {
-    if (!name.startsWith('@aws-cdk/')) {
+    if ((!name.startsWith('@aws-cdk/')) && (!name.startsWith('@aws-solutions-konstruk/'))) {
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/private/runtime-info.ts:19]: Can we move these to an array?

**Comment by hnishar** [comment]
eladb@ added the unit test case and tweaked the code to use a white list for checking which npm modules to add for version reporting.

File: packages/@aws-cdk/core/lib/private/runtime-info.ts [hunk 2]
```diff
@@ -14,9 +16,16 @@ export function collectRuntimeInformation(): cxschema.RuntimeInfo {
     }
   }
 
-  // include only libraries that are in the @aws-cdk npm scope
+  // include only libraries that are in the whitelistLibraries list
   for (const name of Object.keys(libraries)) {
-    if (!name.startsWith('@aws-cdk/')) {
+    let foundMatch = false;
+    whitelistLibraries.forEach(lib => {
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/private/runtime-info.ts:22]: We try to avoid `forEach`. Use a normal `for` loop please.

File: packages/@aws-cdk/core/test/@aws-solutions-konstruk/foo/package.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
+{
```
Comment (eladb) [inline packages/@aws-cdk/core/test/@aws-solutions-konstruk/foo/package.json:1]: Do we really need all of this here? I think we can just do with a tiny `package.json` and `index.js` files.

File: packages/@aws-cdk/core/test/test.app.ts [hunk 4]
```diff
@@ -301,6 +301,31 @@ export = {
     test.done();
   },
 
+  'version reporting includes @aws-solutions-konstruk libraries'(test: Test) {
+    const response = withApp({ runtimeInfo: true }, app => {
+      // eslint-disable-next-line @typescript-eslint/no-require-imports
+      const resolved = require('path').resolve('test/@aws-solutions-konstruk/foo');
```
Comment (eladb) [inline packages/@aws-cdk/core/test/test.app.ts:307]: Why is this needed? Why not just `require('./@aws-solutions-konstruk/foo')`?

Comment (eladb) [inline packages/@aws-cdk/core/test/test.app.ts:307]: Please move this from `test/@aws-solutions-konstruk/foo` to simply `test/runtime-info-konstruk-fixture` (the directory name does not have to be the same as the module name).

File: packages/@aws-cdk/core/test/test.app.ts [hunk 5]
```diff
@@ -301,6 +301,31 @@ export = {
     test.done();
   },
 
+  'version reporting includes @aws-solutions-konstruk libraries'(test: Test) {
```
Comment (eladb) [inline packages/@aws-cdk/core/test/test.app.ts:304]: 1. Extract this to a separate file `test.runtime-info.ts`
2. No need to actually run the entire app. `collectRuntimeInfo` is exported so you should be able to simply call it directly.

**Comment by hnishar** [comment]
eladb@ addressed all the feedback, please review

File: packages/@aws-cdk/core/lib/private/runtime-info.ts [hunk 6]
```diff
@@ -14,9 +17,16 @@ export function collectRuntimeInformation(): cxschema.RuntimeInfo {
...
-  // include only libraries that are in the @aws-cdk npm scope
+  // include only libraries that are in the whitelistLibraries list
   for (const name of Object.keys(libraries)) {
-    if (!name.startsWith('@aws-cdk/')) {
+    let foundMatch = false;
+    for (const lib of WHITELIST_SCOPES) {
+      if (name.startsWith(lib)) {
+        foundMatch = true;
+      }
+    }
```
Comment (eladb) [inline packages/@aws-cdk/core/lib/private/runtime-info.ts:27]:
```suggestion
    for (const scope of WHITELIST_SCOPES) {
      if (name.startsWith(`${scope}/`)) {
        foundMatch = true;
      }
    }
```

**Comment by hnishar** [comment]
@eladb Made some minor updates to unit test and exclude konstruk-fixture/package.json from align-version.sh, please review

**Comment by hnishar** [comment]
@eladb please approve one more time, thanks for your help!
